OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java

```diff
@@ -359,9 +360,10 @@ public boolean isMainUnitLoaded() {
 
     @Override
     public boolean showGoogleCastButton() {
-        if (pager2 != null && pagerAdapter != null) {
-            CourseUnitFragment fragment = ((CourseUnitFragment) pagerAdapter.getItem(pager2.getCurrentItem()));
-            return fragment != null && fragment.hasCastSupportedVideoContent();
+        if(selectedUnit != null && selectedUnit instanceof VideoBlockModel){
+            // Showing casting button only for native video block
+            // Currently casting for youtube video isn't available
+            return ((VideoBlockModel) selectedUnit).getData().encodedVideos.getYoutubeVideoInfo() == null;
```

Comment:
there is the possibility that the course component has youtube videos along with other video encodings.